spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -341,37 +361,51 @@ public void run() {
 				else {
 					for (Message<T> message : messages) {
 						do {
-							try {
-								if (this.listener instanceof PulsarAcknowledgingMessageListener) {
-									this.listener.received(this.consumer, message, this.ackMode.equals(AckMode.MANUAL)
-											? new ConsumerAcknowledgment(this.consumer, message) : null);
-								}
-								else if (this.listener != null) {
-									this.listener.received(this.consumer, message);
-								}
-								if (this.ackMode.equals(AckMode.RECORD)) {
-									handleAck(message);
-								}
-								inRetryMode.compareAndSet(true, false);
+							Observation observation;
```

Comment:
**[NOTE]**: we are only observing single listener (aka record observations) currently. Same approach that spring-kafka has taken. We can look at how to add that to the batch case sensibly once the dust settles on this basic use case.

Reply:
Will handle/consider in #149